File: examples/src/main/scala/examples/nodejs/general/TimersExample.scala
package examples.nodejs.general

import org.scalajs.nodejs._
import org.scalajs.nodejs.Bootstrap

import scala.concurrent.duration._

/**
  * Timers Example
  * @author lawrence.daniels@gmail.com
  */
class TimersExample(bootstrap: Bootstrap) {

  setImmediate(() => console.log("Hello"))

  setTimeout(() => console.log("\tWorld"), 0.5.second)

  setTimeout(() => console.log("\t\tfrom MEANS.js"), 1.second)

}

File: examples/src/main/javascript/nodejs/general/TimersExample.js
/**
 * Timers Example
 * @author lawrence.daniels@gmail.com
 */
(function() {

    setImmediate(function () { console.log("Hello") });

    setTimeout(function () { console.log("\tWorld") }, 500);

    setTimeout(function () { console.log("\t\tfrom MEANS.js") }, 1000);
    
})();